crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -165,6 +172,28 @@ where
                     .storage_as_mut::<OwnedCoins>()
                     .remove(&key)?;
             }
+            Event::ForcedTransactionFailed {
+                id,
+                block_height,
+                block_time,
+                failure,
+            } => {
+                let status = RelayedTransactionStatus::Failed {
+                    block_height: *block_height,
+                    block_time: *block_time,
+                    failure: failure.clone(),
+                };
+
+                block_st_transaction
+                    .storage_as_mut::<RelayedTransactionStatuses>()
+                    .insert(&Bytes32::from(id.to_owned()), &status)?;
+            }
+            _ => {
```

Comment:
We probably don't need to make this non-exhaustive since changing this enum won't really work in a forkless upgrade anyways